### PR TITLE
Fixed nicole branch

### PR DESCRIPTION
**Pull request by NicoleKMarino**
Multi-tenancy authorization, can create an employer account

### CODE REVIEW

**Comment by NicoleKMarino** [comment]
This functionality allows a user to sign up as a regular user or as an employer. It sets up multi-tenancy and authorizes based on controller. Redirects to different dashboards based on roles. 
@jmejia- Is my method in the user controller called "set_role" explicit and readable for other developers, or is there a way I can refactor?

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -3,23 +3,27 @@ def login_referrer
...
+  def set_user
+    # if params[:commit]
+      @user = User.find_by(username: params[:session][:username])
+    # else
+    #   @user = User.from_omniauth(request.env["omniauth.auth"])
+    # end
+  end
+
   def redirect_based_on_referrer
     if login_referrer == 'cart'
```
Comment (roscalabrin) [inline app/helpers/sessions_helper.rb:16]: @NicoleKMarino: this should be updated to 'bucket' instead of cart

File: app/models/user.rb [hunk 2]
```diff
@@ -27,4 +27,17 @@ def self.from_omniauth(auth_info)
   def send_welcome_email
     UserNotifierMailer.send_signup_email(self).deliver if self.email
   end
+
+
+    def platform_admin?
```
Comment (roscalabrin) [inline app/models/user.rb:32]: @NicoleKMarino: these methods look great. Just need to fix the indentation

File: app/services/permission_service.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
+require 'pry'
+class PermissionService
+  attr_reader :user
+
+  def initialize(user)
+    @user= user || User.create
+  end
+
+  def allow?(controller)
```
Comment (roscalabrin) [inline app/services/permission_service.rb:9]: @NicoleKMarino we can refactor this method pulling out the conditionals into separate methods (remember what we did in class with Jorge?)

**Pull request closed without merging**